Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -334,7 +336,312 @@ void reverse_face_orientations(const FaceRange& face_range, PolygonMesh& pmesh)
+    }
+
+    internal::recursive_orient_volume_ccs<Kernel>(
+           tm, vpm, fid_map, xtrm_vertices, new_cc_handled, face_cc,
+           new_xtrm_cc_id, new_is_parent_outward_oriented);
+  }
+
+// now explore remaining cc included in the same cc as xtrm_cc_id
+  boost::dynamic_bitset<> cc_not_handled = ~cc_handled;
+  std::size_t new_xtrm_cc_id = cc_not_handled.find_first();
+  if (new_xtrm_cc_id == cc_not_handled.npos) return ;
+
+  for (std::size_t candidate = cc_not_handled.find_next(new_xtrm_cc_id);
+                   candidate < cc_not_handled.npos;
+                   candidate = cc_not_handled.find_next(candidate))
+  {
+     if(get(vpm,xtrm_vertices[candidate]).z() > get(vpm,xtrm_vertices[new_xtrm_cc_id]).z())
+        new_xtrm_cc_id = candidate;
+  }
+
+  internal::recursive_orient_volume_ccs<Kernel>(
+            tm, vpm, fid_map, xtrm_vertices, cc_handled, face_cc,
+            new_xtrm_cc_id, is_parent_outward_oriented);
+}
+
+}//end internal
+
+/**
+* \ingroup PMP_orientation_grp
+* makes each connected component of a closed triangulated surface mesh
+* inward or outward oriented.
+*
+* @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph` .
+*                      If `TriangleMesh` has an internal property map for `CGAL::face_index_t`,
+*                      as a named parameter, then it must be initialized.
+* @tparam NamedParameters a sequence of \ref namedparameters
+*
+* @param tm a closed triangulated surface mesh
+* @param orient_outward `true` indicates that each connected component will be outward oriented,
+*                        (inward oriented if `false`). Default value is true.
+* @param np optional sequence of \ref namedparameters among the ones listed below
+*
+* \cgalNamedParamsBegin
+*   \cgalParamBegin{vertex_point_map}
+*     the property map with the points associated to the vertices of `tm`.
+*     If this parameter is omitted, an internal property map for
+*     `CGAL::vertex_point_t` should be available in `TriangleMesh`
+*   \cgalParamEnd
+*   \cgalParamBegin{face_index_map}
+*     a property map containing the index of each face of `tm`.
+*   \cgalParamEnd
+* \cgalNamedParamsEnd
+*/
+template<class TriangleMesh, class NamedParameters>
+void orient(TriangleMesh& tm, bool orient_outward, const NamedParameters& np)
+{
+  typedef boost::graph_traits<TriangleMesh> Graph_traits;
+  typedef typename Graph_traits::vertex_descriptor vertex_descriptor;
+  typedef typename Graph_traits::face_descriptor face_descriptor;
+  typedef typename Graph_traits::halfedge_descriptor halfedge_descriptor;
+  typedef typename GetVertexPointMap<TriangleMesh,
+      NamedParameters>::const_type Vpm;
+  typedef typename GetFaceIndexMap<TriangleMesh,
+      NamedParameters>::const_type Fid_map;
+
+  CGAL_assertion(is_triangle_mesh(tm));
+  CGAL_assertion(is_valid(tm));
+  CGAL_assertion(is_closed(tm));
+
+  Vpm vpm = boost::choose_param(get_param(np, internal_np::vertex_point),
```

Comment:
don't you need the namespace `boost` for `get_param()` and `get_const_property_map()`?